docs/standard/serialization/system-text-json-how-to.md

```diff
@@ -475,7 +475,7 @@ In this scenario, the `WindSpeed` property is not serialized even if the `weathe
 
```

Comment:
I think we should add a blurb like the following to the beginning of this heading to be explicit about this (`Serialize properties of derived classes`). Otherwise, folks might be mislead into thinking the scenario is supported:

Serialization of deep, strongly-typed polymorphic type heirarchy within an object graph is **not** supported. This includes interfaces and abstract classes where the runtime types differ from the statically defined types declared on the object. One exception to this is properties of the `System.Object` type itself.

And then we can later showcase some examples that won't work (like we already have `WeatherForecastDerived`).

We should consider adding one "real" example for interfaces too. Here's an example to showcase the point:
```C#
public interface IMyInterface
{
    public int Count { get; set; }
}

public class MyImplementation : IMyInterface
{
    public int Count { get; set; }
    public string Name { get; set; }
}

public class PolymorphicTestInterface
{
    public IMyInterface MyObject { get; set; } // Change this to object to get the desired behavior
}

var myImpl = new MyImplementation
{
    Count = 5,
    Name = "Implementation"
};

var polymorphicObject = new PolymorphicTestInterface
{
    MyObject = myImpl
};

// {"MyObject":{"Count":5}}
// Name is not serialized
Console.WriteLine(JsonSerializer.Serialize(polymorphicObject));
```